libtransmission/torrents.cc

```diff
@@ -17,6 +18,45 @@
 namespace
 {
 
+struct CompareTorrentByHashLen
```

Comment:
This class + the building of the partial `tr_sha1_digest_t` seems like extra complexity + work.

I might be wrong but I _think_ we could get by with a Compare that takes `tr_torrent const*` + the passed into `tr_torrents::get()` and do comparisons with `tor->infoHashString()`, since those values are just as sorted as `tor->infoHash()` is. _If_ that works, it would simplify the code.

Something like

```c++
bool operator(tr_torrent* const tor, std::string_view info_hash_prefix) const
{
    auto const n = std::size(info_hash_prefix);
    auto const tor_info_hash_prefix = std::string_view{ tor->infoHashString() }.substr(0, n);
    return tor_info_hash_prefix < info_hash_prefix;
}
```

Reply:
I mainly wanted to avoid the multiple calls to `infoHashString()`. I guess it is a trade off of performance versus code complexity.

Reply:
calls to `infoHashString()` is [a `constexpr` function that just returns a const reference to a private field](https://github.com/transmission/transmission/blob/2866638e1b3672c897692b7fb10c8e17354ee05c/libtransmission/magnet-metainfo.h#L57), so it should essentially be free.

By using a string_view for `tor_info_hash_prefix` you don't have to alloc or copy any chars to make that temporary substring, either. :tada:

Reply:
I changed as you suggested, so now the code is much simpler, but the performance is degraded, though it is probably unnoticeable. This also fixes the .value problem.